twitch/connection.go

```diff
@@ -25,15 +25,18 @@ type IRC struct {
 	client   *v2.Client
 	wg       *sync.WaitGroup
 	tok      *oauth2.Token
+	msgQueue chan string
 }
 
 // SetupTwitchIRC sets up the IRC, configures oauth, and inits connection functions.
 func SetupTwitchIRC(db postgres.PG, wg *sync.WaitGroup) (*IRC, error) {
 	irc := &IRC{
 		database: db,
 		wg:       wg,
+		msgQueue: make(chan string),
 	}
 	wg.Add(1)
+	defer wg.Done()
```

Comment:
same thing with this defer. WG.Done()  will fire as soon as the routine is launched.